spec/acceptance/fail2ban_spec.rb

```diff
@@ -0,0 +1,237 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+case fact('os.family')
+when 'Debian'
+  package_name          = 'fail2ban'
+  config_file_path      = '/etc/fail2ban/jail.conf'
+  service_name          = 'fail2ban'
+  ssh_log_file          = '/var/log/auth.log'
+  ssh_jail              = 'sshd'
+when 'RedHat'
+  package_name          = 'fail2ban'
+  config_file_path      = '/etc/fail2ban/jail.conf'
+  service_name          = 'fail2ban'
+  ssh_log_file          = '/var/log/secure'
+  ssh_jail              = 'ssh'
+end
+
+# Ensure the ssh log file is created, otherwise the service doesn't start completely
+shell("touch #{ssh_log_file}")
+
+describe 'fail2ban' do
+  context 'with defaults' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      class { 'fail2ban': }
+        EOS
+      end
+    end
+
+    describe package(package_name) do
+      it { is_expected.to be_installed }
+    end
+
+    describe file(config_file_path) do
+      it { is_expected.to be_file }
+    end
+
+    describe service(service_name) do
+      it { is_expected.to be_running }
+      it { is_expected.to be_enabled }
+    end
+
+    describe command('fail2ban-client status') do
+      its(:stdout) { is_expected.to match %r{Jail list:\s+#{ssh_jail}} }
+    end
+  end
+
+  context 'when package absent' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      class { 'fail2ban':
+        package_ensure => 'absent',
+        service_ensure => 'stopped',
+        service_enable => false,
+      }
+        EOS
+      end
+    end
+
+    describe package(package_name) do
+      it { is_expected.not_to be_installed }
+    end
+
+    describe file(config_file_path) do
+      it { is_expected.to be_file }
+    end
+
+    describe service(service_name) do
+      it { is_expected.not_to be_running }
+      it { is_expected.not_to be_enabled }
+    end
+  end
+
+  context 'when content template' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-EOS
+      $_config_file_template = $facts['os']['family'] ? {
```

Comment:
Same issue with outdenting.  This would also be better as an `example/*.pp` so that it is linted.